src/java/src/main/native/utils.h

```diff
@@ -30,19 +31,24 @@ bool ThrowIfError(JNIEnv* env, OgaResult* result);
 // handle conversion/release of jstring to const char*
 struct CString {
   CString(JNIEnv* env, jstring str)
-      : env_{env}, str_{str}, cstr{env->GetStringUTFChars(str, /* isCopy */ nullptr)} {
+      : env_{env}, str_{str}, cstr{env->GetStringUTFChars(str, /* isCopy */ nullptr)}, len_{env->GetStringUTFLength(str)} {
   }
 
   const char* cstr;
 
   operator const char*() const { return cstr; }
 
+  std::string utf8String() {
```

Comment:
Strictly speaking, this is Java modified version of UTF8. It'll work for ASCII. But I'm not sure how good it is comparing to "normal" UTF8.